frontend/alert/components/GroupSuggestion.tsx

```diff
@@ -0,0 +1,357 @@
+            } else if (top < parentTop) {
+                parentElement.scrollBy({ top: top - parentTop });
+            }
+        }
+    }, [selected, ref]);
+
+    return (
+        <DropdownItemBox selected={selected} ref={ref}>
+            <CheckboxContainer
+                onClick={() => {
+                    onClick();
+                    setChecked(!checked);
+                }}
+            >
+                <Checkbox checked={checked} />
+            </CheckboxContainer>
+            <DropdownItemLeftCol
+                onClick={() => {
+                    onClick();
+                    setChecked(!checked);
+                }}
+            >
+                <SuggestionTitle>{courseCode}</SuggestionTitle>
+                <SuggestionSubtitle>{instructor || "TBA"}</SuggestionSubtitle>
+            </DropdownItemLeftCol>
+            <IconContainer onClick={onChangeTimeline}>
+                <HistoryIcon icon={faHistory} className="historyIcon" />
+            </IconContainer>
+        </DropdownItemBox>
+    );
+};
+
+interface GroupSuggestionProps {
+    sections: any;
+    courseCode: string;
+    value: string;
+    selectedCourses: Set<Section>;
+    inputRef: React.RefObject<HTMLInputElement>;
+    setValue: (v: any) => void;
+    setTimeline: React.Dispatch<React.SetStateAction<string | null>>;
+    setSelectedCourses: any;
+}
+
+const GroupSuggestion = ({
+    sections,
+    courseCode,
+    value,
+    selectedCourses,
+    inputRef,
+    setValue,
+    setTimeline,
+    setSelectedCourses,
+}: GroupSuggestionProps) => {
+    const initializeButtonStates = () => {
+        const states = {};
+        Object.keys(sections).map((key) => {
+            states[key] = false;
+        });
+
+        return states;
+    };
+
+    const [buttonStates, setButtonStates] = useState(initializeButtonStates);
+
+    /**
+     * Takes the activity/activity of a section and return if all sections of {activity} have been selected
+     * Used to make sure to update button states when manually selecting each section
+     * @param activity - section activity
+     */
+    const selectedAllActivity = (activity) => {
+        // check if the user selected all courses of a certain activity in a group suggestion
+        let selectedAll = true;
+
+        sections[activity].map((section) => {
+            if (!selectedCourses.has(section)) {
+                selectedAll = false;
+            }
+        });
+
+        syncButtonStates(activity, selectedAll);
+        return selectedAll;
+    };
+
+    /**
+     * Update button states to match selected courses in the case the user is manually checking/unchecking sections
+     * Ex. allLectures should be true when user manually select all "LEC" sections
+     * @param activity - section activity
+     */
+    const syncButtonStates = (activity, correctState) => {
+        // update button state if its not the same as correctState
+        if (buttonStates[activity] != correctState) {
```

Comment:
I think `!==` is considered better practice

Reply:
Right!!